homeassistant/components/template/lock.py

```diff
@@ -120,7 +120,7 @@ def _update_state(self, result):
         if isinstance(result, TemplateError):
             self._state = None
             return
-        self._state = result.lower() in ("true", STATE_ON, STATE_LOCKED)
+        self._state = str(result).lower() in ("true", STATE_ON, STATE_LOCKED)
```

Comment:
If we know we're going to convert it back to a string, should we still parse it?

Oh I guess we should, because when the render result is `"hello"`, it will create `hello`. This can also be a pretty big breaking change. I think that we should make this option initially opt-in, and later opt-out.

Reply:
No, this more a lame simplification. The result can be a boolean, but also "ON", "on", "locked" "LOCKED". By converting it to a string, it makes the test easier in code.

The extensive version of this, would be checking if this is a boolean type -> use as state. If it is a string type, lower() in (STATE_ON, STATE_LOCKED). Any other type -> state = Unknown?

Reply:
Looking at the latter proposal, while that is IMHO the correct way to implement this, is a breaking change. Examples `hello`, ``, `none`, as a template result will now show locked.

While not the prettiest, it is the most compatible situation right now.

Reply:
Improved the lock & switch state logic in https://github.com/home-assistant/core/pull/41227/commits/e878e29c95908ef846849f0bd61b99f08ae4ecfb